torch/csrc/jit/tensorexpr/analysis.h

```diff
@@ -46,6 +47,102 @@ class NodeFinder : public IRVisitor {
   std::vector<Node*> nodes;
 };
 
+// A class that analyzes the given program relevant for Block backend.
+class BlockAnalysis : public IRVisitor {
+ public:
+  bool is_buf_store_target(const Buf* buf) const {
+    return store_targets_.count(buf) > 0;
+  }
+
+  std::unordered_set<const Buf*> reads() const {
+    return loads_;
+  }
+
+  std::unordered_set<const Buf*> stores() const {
+    return store_targets_;
+  }
+
+  int block_size() const {
+    return block_size_;
+  }
+
+  void analysis_ran() {
+    analysis_runs_++;
+    return;
+  }
+
+  bool areBufsInMap(const std::unordered_set<const Buf*>& bufs) const {
+    for (auto const& arg : bufs) {
```

Comment:
I suggest moving the implementation to a cpp file to keep header files lean.

Reply:
Done!